RUM mobile SDK docs restructure

rtrieu opened the pull request:
<!-- *Note: Please remember to review the Datadog Documentation [Contribution Guidelines](https://github.com/DataDog/documentation/blob/master/CONTRIBUTING.md) if you have not yet done so.* -->

### What does this PR do? What is the motivation?
<!-- A brief description of the change being made with this pull request. What is your motivation for the PR? -->
As per [DOCS-9231](https://datadoghq.atlassian.net/browse/DOCS-9231):

Restructures RUM Mobile docs so that they are organized by framework instead of by topic
### Merge instructions

<!-- 
If you're waiting for a release or there are other considerations that you want us to be aware of, list them here. 
If the PR is ready to be merged once it recieves the required reviews, check the box below after you've created the PR.
-->

Merge readiness:
- [ ] Ready for merge

Merge queue is enabled in this repo. To have it automatically merged after it receives the required reviews, create the PR (from a branch that follows the `<yourname>/description` naming convention) and then add the following PR comment:

```
/merge
```

### Additional notes
<!-- Anything else we should know when reviewing?-->

<!-- Previewing the PR: Assuming you are a Datadog employee and named your branch `<yourname>/<description>`, a preview build will run and links to the preview output will be auto-generated and posted in the PR comments. The links will 404 until the preview build is finished running. -->


[DOCS-9231]: https://datadoghq.atlassian.net/browse/DOCS-9231?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

jhgilbert commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/roku/_index.md:
```diff
@@ -0,0 +1,32 @@
+To get started with RUM for Roku, create an application and configure the RokuSDK.
+
+{{< whatsnext desc="This section includes the following topics:">}}
+  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/roku/setup">}}<u>Setup</u>: Learn how to setup the Roku SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```
```suggestion
  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/roku/setup">}}<u>Setup</u>: Learn how to set up the Roku SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```

jhgilbert commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/android/_index.md:
```diff
@@ -0,0 +1,35 @@
+To get started with RUM for Android, create an application and configure the Android SDK.
+
+{{< whatsnext desc="This section includes the following topics:">}}
+  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/android/setup">}}<u>Setup</u>: Learn how to setup the Android SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```
```suggestion
  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/android/setup">}}<u>Setup</u>: Learn how to set up the Android SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```

jhgilbert commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/flutter/_index.md:
```diff
@@ -0,0 +1,32 @@
+To get started with RUM for Flutter, create an application and configure the Flutter SDK.
+
+{{< whatsnext desc="This section includes the following topics:">}}
+  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/flutter/setup">}}<u>Setup</u>: Learn how to setup the Flutter SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```
```suggestion
  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/flutter/setup">}}<u>Setup</u>: Learn how to set up the Flutter SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```

jhgilbert commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/ios/_index.md:
```diff
@@ -0,0 +1,34 @@
+To get started with RUM for iOS, create an application and configure the iOS SDK.
+
+{{< whatsnext desc="This section includes the following topics:">}}
+  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/ios/setup">}}<u>Setup</u>: Learn how to setup the iOS SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```
```suggestion
  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/ios/setup">}}<u>Setup</u>: Learn how to set up the iOS SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```

jhgilbert commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/_index.md:
```diff
@@ -0,0 +1,36 @@
+To get started with RUM for Kotlin Multiplatform, create an application and configure the Kotlin Multiplatform SDK.
+
+{{< whatsnext desc="This section includes the following topics:">}}
+  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup">}}<u>Setup</u>: Learn how to setup the Kotlin Multiplatform SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```
```suggestion
  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup">}}<u>Setup</u>: Learn how to set up the Kotlin Multiplatform SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```

jhgilbert commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/react_native/_index.md:
```diff
@@ -0,0 +1,36 @@
+To get started with RUM for React Native, create an application and configure the React Native SDK.
+
+{{< whatsnext desc="This section includes the following topics:">}}
+  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/react_native/setup">}}<u>Setup</u>: Learn how to setup the React Native SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```
```suggestion
  {{< nextlink href="/real_user_monitoring/mobile_and_tv_monitoring/react_native/setup">}}<u>Setup</u>: Learn how to set up the React Native SDK, track background events, and send data when devices are offline.{{< /nextlink >}}
```

jhgilbert commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/unity/data_collected.md:
```diff
@@ -1,11 +1,9 @@
 ---
-title: RUM Unity Data Collected
```
I think Data Collected still needs to be added under Unity in the nav, see [the staging page](https://docs-staging.datadoghq.com/rtrieu/rum-mobile-sdk-docs-restructure/real_user_monitoring/mobile_and_tv_monitoring/unity/data_collected/)

jhgilbert commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/unity/advanced_configuration.md:
```diff
@@ -1,15 +1,13 @@
 ---
-title: RUM Unity Advanced Configuration
+title: Unity Advanced Configuration
```
I think Advanced Configuration still needs to be added under Unity in the nav, see [the staging page](https://docs-staging.datadoghq.com/rtrieu/rum-mobile-sdk-docs-restructure/real_user_monitoring/mobile_and_tv_monitoring/unity/advanced_configuration/)

jhgilbert reviewed (commented):
Congrats for making all of this happen, this is a major reorg! 

In addition to the suggestions, I also found that two of the staging pages look a bit odd in that they don't have a corresponding nav entry. Were they supposed to be deleted / become aliases instead?

- [Web Tracking](https://docs-staging.datadoghq.com/rtrieu/rum-mobile-sdk-docs-restructure/real_user_monitoring/mobile_and_tv_monitoring/web_view_tracking/?tab=android)
- [Android Crash Reporting and Error Tracking](https://docs-staging.datadoghq.com/rtrieu/rum-mobile-sdk-docs-restructure/error_tracking/frontend/mobile/android/?tab=us)

Disregard if not, thank you!

rtrieu commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/android/integrated_libraries.md:
```diff
@@ -1,15 +1,13 @@
   tag: "Source Code"
   text: Source code for dd-sdk-android
-- link: /real_user_monitoring/mobile_and_tv_monitoring/integrated_libraries
+- link: /real_user_monitoring/mobile_and_tv_monitoring/android/integrated_libraries
```
removed, thank you!

rtrieu commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/unity/setup.md:
```diff
@@ -23,9 +21,9 @@ further_reading:
-Unity Monitoring is in Preview.
-{{< /callout >}}
+{{< beta-callout url="#" btn_hidden="true" >}}
+Unity Monitoring is in public beta.
```
This was not intentional :) reverting! Thanks for the catch.

rtrieu commented:
Hi @jhgilbert, thank you for catching all the things you did in the last review, which I believe I've addressed! Can you give it another pass? Here are some significant new updates:
- Expo and CodePush are no longer part of the React Native multi-code-lang pages, and are now their own sections with reused content
- Added link to the specific tabs for platforms on the Mobile Vitals and Web View Tracking pages
- Mobile Vitals has been removed for Roku and Unity, as it's not relevant for those platforms
- Reconciled some missing info from the Unity setup doc

jhgilbert commented:
@rtrieu Thanks for the bump! All of the pages listed under removed files in the preview links are 404s ... should they redirect?

rtrieu commented:
@jhgilbert Good call! Just added redirects for those. Also, I had to take back my update to put CodePush and Expo their own sections! They are now back nested with React Native.

rtrieu commented:
/merge

rtrieu commented:
/merge -c

StefonSimmons commented on layouts/partials/rum/rum-react-native-setup.html:
@rtrieu I dont see where this partial is being used. where is it referenced

rtrieu replied:
Ah, I need to delete that. Previously, I had it here: content/en/real_user_monitoring/mobile_and_tv_monitoring/react_native/setup/_index.md

StefonSimmons commented on content/en/real_user_monitoring/platform/connect_rum_and_traces.md:
```diff
@@ -283,8 +283,8 @@ To start sending just your iOS application's traces to Datadog, see [iOS Trace C
 
-[1]: /real_user_monitoring/mobile_and_tv_monitoring/setup/flutter/
-[2]: /real_user_monitoring/mobile_and_tv_monitoring/setup/flutter/#automatic-resource-tracking
+[1]: /real_user_monitoring/mobile_and_tv_monitoring/flutter/etup/
```
```suggestion
[1]: /real_user_monitoring/mobile_and_tv_monitoring/flutter/setup/
```

StefonSimmons commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup.md:
```diff
@@ -112,30 +110,27 @@ If you are integrating Kotlin Multiplatform library as a framework with an `embe
 {{% tab "RUM" %}}
 
-1. Navigate to [**Digital Experience** > **Add an Application**][1].
+1. Navigate to [**Digital Experience** > **Add an Application**][2].
```
@rtrieu does this have the correct link?

rtrieu replied:
Updated to [1], thanks!

rtrieu commented on content/en/real_user_monitoring/mobile_and_tv_monitoring/kotlin_multiplatform/setup.md:
```diff
@@ -112,30 +110,27 @@ If you are integrating Kotlin Multiplatform library as a framework with an `embe
 {{% tab "RUM" %}}
 
-1. Navigate to [**Digital Experience** > **Add an Application**][1].
+1. Navigate to [**Digital Experience** > **Add an Application**][2].
```
```suggestion
1. Navigate to [**Digital Experience** > **Add an Application**][1].
```

rtrieu replied:
```suggestion
1. Navigate to [**Digital Experience** > **Add an Application**][1].
```

rtrieu replied:
```suggestion
1. Navigate to [**Digital Experience** > **Add an Application**][1].
```

rtrieu commented:
/merge
